build-scripts/multidirectory.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+# Creates/removes/updates branches according the `jobs/` directory on the
+# master branch.  Local files and branches will be modified, so this script
+# should be executed on a clean, dedicated clone.
+set -eu
+
+set_difference() {
+    printf '%s\n%s\n%s\n' "$1" "$1" "$2" | sort | uniq -u
+}
+
+list_branches() {
+    git branch --list --remotes 'origin/*' \
```

Comment:
It's a little hard for me to decode this, but I think you're taking a list of refs like `origin/branchname` and getting just `branchname`?

Always prefer to write non-Bash like Bash:

```sh
# list_branches lists the branch names for the `upstream' remote
function list_branches() {
    for branch in $( git branch --list --remotes ); do
        if [[ "${branch}" == upstream/* ]]; then # how does your script handle entries like upstream/HEAD -> upstream/master
            echo "${branch:9}" # if string slicing isn't your thing, could use sed
        fi
    done
}
```